internal/domain/user.go

```diff
@@ -0,0 +1,136 @@
+package domain
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
+)
+
+type UserAggregate struct {
+	id                    uuid.UUID `validate:"required"`
+	Username              string    `validate:"required,min=2,max=50"`
+	Email                 string    `validate:"required,email"`
+	isVerified            bool
+	verifiedAt            *time.Time
+	lastPasswordChangedAt *time.Time
+	registeredAt          time.Time `validate:"required"`
+	hashedPassword        []byte    `validate:"required"`
+	isSuperUser           bool
+}
+
+func (u *UserAggregate) GetIdentifier() uuid.UUID {
+	return u.id
+}
+
+func (u *UserAggregate) Validate() error {
+	valdidate := validator.New()
+	return valdidate.Struct(u)
+}
+
+func (u *UserAggregate) ChangedEmail(newEmail string) error {
```

Comment:
I wonder; you've defined previously a separate type for Environments, but you are not doing it for an Email; how do you decide what should be a custom type and what should be a primitive type?

Reply:
I did it on purpose because I wanted to ask you if there are any libraries in Go that already contain such types? : "Email, Password / SecretStr?"